### DISCUSSION TITLE
Change php.ini

### DISCUSSION THREAD

**Discussion by fireblade2534** [nextcloud/all-in-one#4748] [Questions] [answered]
I would like to modify php.ini and have the changes persist. How do I do that?

**Comment by szaimen** [comment]
Hi, why do you want to do this?

**Reply by fireblade2534** [reply]
I’m trying to optimize my instance and a lot of the tutorials say to change things in php.ini

**Reply by szaimen** [reply]
What values do you want to adjust?

**Reply by fireblade2534** [reply]
Stuff like
opcache.jit = 1255
opcache.jit_buffer_size = 128M

**Reply by szaimen** [reply, marked as answer]
You don't need this as AIO applies this automatically for you: https://github.com/search?q=repo%3Anextcloud%2Fall-in-one%20opcache.jit%20&type=code

**Reply by sasikora** [reply]
Hello there,

so we changed:
session.cache_limiter = nocache
to
session.cache_limiter = public

It improved the overall performance a lot, especially for streaming/viewing some high-quality files.
We had this running for a week without any issues but I imagine not everyone wants this to apply but we would really appreciate it if you could make this persistent/configurable. Ideally as an env parameter in the aio docker-compose file.

To make it clean "session.cache_expire" should also be added. Thanks in advance!

**Reply by szaimen** [reply]
> so we changed:
> session.cache_limiter = nocache
> to
> session.cache_limiter = public

Not sure if we are going to do this. See https://github.com/nextcloud/server/pull/46320/files


> To make it clean "session.cache_expire" should also be added.

why?

**Reply by sasikora** [reply]
> Not sure if we are going to do this. See https://github.com/nextcloud/server/pull/46320/files

So if I understood correctly it will (maybe, if it gets picked up again) be changed if the checks succeed to overwrite to '' e.g. "no-preference" at runtime right? This might also help as the current value "nocache" explictly tells everything to forbid caching.
I will configure it to '' in our instance and see what it does.

> why?

Because (I think, but quite sure) it's the corresponding value to the "session.cache_limiter" which tells all the various caches (proxy and client) when to drop the stored data again.
